### PR TITLE
Foreground layers - (Images & Videos)

### PR DESCRIPTION
**Pull request by maddygoround**

### CODE REVIEW

File: src/compositor/compositor.ts [hunk 1]
```diff
@@ -358,7 +358,7 @@ export const start = (settings: Settings): CompositorInstance => {
           // Trigger compositor-only event
           triggerEvent('NodeRemoved', {
             projectId: project.id,
-            nodeId: nodeIndex.id,
+            nodeId: nodeIndex[id].id,
```
Comment (maddygoround) [inline src/compositor/compositor.ts:361]: `nodeIndex.id `is undefined. The value if part of `nodeIndex[id].id`

File: src/compositor/html/html-transforms.ts [hunk 2]
```diff
@@ -187,6 +187,7 @@ export const init = (
           onEvent: compositor.on,
           onNewSource: (cb) => _onNewSourceHandlers.push(cb),
           onUpdate: (cb) => _onUpdateHandlers.push(cb),
+          nodeId : node.id
```
Comment (maddygoround) [inline src/compositor/html/html-transforms.ts:190]: Node Id added to transform context. This make nodeId available in every **create** function context element.

File: src/compositor/transforms.ts [hunk 3]
```diff
@@ -55,13 +56,15 @@ export type TransformElementGetter = (node: SceneNode) => TransformElement
 export type TransformContext = {
   trigger: (event: string, payload: any) => void
   /** Listens for all events emitted by the compositor */
-  onEvent?: (name: string, payload: any) => void
+  onEvent?: (event: string, cb: (payload: any) => void, nodeId?: string) => Disposable
```
Comment (maddygoround) [inline src/compositor/transforms.ts:59]: onEvent function definition changed to match compositor.on

File: src/core/transforms/Video.ts [hunk 4]
```diff
@@ -21,25 +23,32 @@ let interval: NodeJS.Timer
 export const Video = {
   name: 'LS-Video',
   sourceType: 'LS-Video',
-  create({ onUpdate, trigger }) {
+  create({ onUpdate, trigger, onEvent, nodeId }) {
+    const disposeThisEvent = onEvent('NodeRemoved', (payload) => {
+      if (payload.nodeId === nodeId) {
+        clearInterval(interval)
```
Comment (maddygoround) [inline src/core/transforms/Video.ts:29]: dispose the values if nodeId matches

File: src/core/transforms/Video.ts [hunk 5]
```diff
@@ -24,12 +24,11 @@ export const Video = {
...
   create({ onUpdate, trigger, onEvent, nodeId }) {
-    const disposeThisEvent = onEvent('NodeRemoved', (payload) => {
-      if (payload.nodeId === nodeId) {
+
+    const disposeThisEvent = onEvent('NodeRemoved', () => {
         clearInterval(interval)
         disposeThisEvent()
-      }
-    })
+    },nodeId)
```
Comment (maddygoround) [inline src/core/transforms/Video.ts:31]: Just figured we can listen on a specific nodeId. Huraay! (Thanks @hudbuddy )

Comment (hudbuddy) [reply src/core/transforms/Video.ts:31]: 🥳

**Pull request closed without merging**